community/bolt/src/test/java/org/neo4j/bolt/v1/transport/integration/TransportSessionIT.java

```diff
@@ -283,9 +288,39 @@ public void shouldFailNicelyOnPoints() throws Throwable
         assertThat( client, eventuallyRecieves(
                 msgSuccess(),
                 msgSuccess( map( "fields", singletonList( "p") ) ),
+                msgRecord(eqRecord( nullValue() )),
                 msgFailure( Status.Request.Invalid, "Point is not yet supported as a return type in Bolt")) );
     }
 
+    @Test
+    public void shouldFailNicelyOnBinary() throws Throwable
+    {
+        //Given
+        GraphDatabaseService db = server.graphDatabaseService();
+        try ( Transaction tx = db.beginTx())
+        {
+            Node node = db.createNode();
+            node.setProperty( "binary", new byte[]{(byte)0xB0, 0x17} );
+            tx.success();
+        }
+
+        // When
+        client.connect( address )
+                .send( TransportTestUtil.acceptedVersions( 1, 0, 0, 0 ) )
+                .send( TransportTestUtil.chunk(
+                        init( "TestClient/1.1", emptyMap() ),
+                        run( "MATCH (n) RETURN n.binary" ),
+                        pullAll() ) );
+
+        // Then
+        assertThat( client, eventuallyRecieves( new byte[]{0, 0, 0, 1} ) );
+        assertThat( client, eventuallyRecieves(
+                msgSuccess(),
+                msgSuccess( map( "fields", singletonList( "n.binary") ) ),
+                msgRecord(eqRecord( nullValue() )),
+                msgFailure( Status.Request.Invalid, "Binary values is not yet supported in Bolt")) );
```

Comment:
"Binary values" is a bit too vague - e.g. booleans are binary values. Why not refer to "byte arrays" when that is what they really are?